src/EditorFeatures/CSharpTest/DocumentationComments/CodeFixes/AddDocCommentParamNodeCodeFixProviderTests.cs

```diff
@@ -0,0 +1,677 @@
+    public class AddDocCommentParamNodeCodeFixProviderTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return new Tuple<DiagnosticAnalyzer, CodeFixProvider>(null, new CSharpAddDocCommentParamNodeCodeFixProvider());
+        }
+
+        private async Task TestAsync(string initial, string expected)
+        {
+            var parseOptions = Options.Regular.WithDocumentationMode(DocumentationMode.Diagnose);
+            await TestAsync(initial, expected, parseOptions: parseOptions, compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NoNodesBefore()
+        {
+            var initial = 
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""j""></param>
+    public void Fizz(int [|i|], int j, int k) {}
+}
+";
+
+            var expected =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""i""></param>
+    /// <param name=""j""></param>
+    public void Fizz(int i, int j, int k) {}
+}
+";
+            await TestAsync(initial, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NoNodesAfter()
+        {
+            var initial =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""j""></param>
+    public void Fizz(int i, int j, int [|k|]) {}
+}
+";
+
+            var expected =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""j""></param>
+    /// <param name=""k""></param>
+    public void Fizz(int i, int j, int k) {}
+}
+";
+            await TestAsync(initial, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NodesBeforeAndAfter()
+        {
+            var initial =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""i""></param>
+    /// <param name=""k""></param>
+    public void Fizz(int i, int [|j|], int k) {}
+}
+";
+
+            var expected =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""i""></param>
+    /// <param name=""j""></param>
+    /// <param name=""k""></param>
+    public void Fizz(int i, int j, int k) {}
+}
+";
+            await TestAsync(initial, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NestedInSummaryTag()
+        {
+            var initial =
+@"class Program
+{
+    /// <summary>
+    /// <param name=""j""></param>
+    /// </summary>
+    public void Fizz(int i, int j, int [|k|]) {}
```

Comment:
I'm probably approaching this the wrong way, but would do I remove all nodes in one sequence of nodes and add a series of nodes in the position of the first node in the sequence of removed nodes?